src/providers/code-actions/commonActions.ts

```diff
@@ -56,3 +60,27 @@ export const createIgnorePathAction = (
   return ignorePathAction;
 };
 
+export const createOpenViolationCardAction = (
+    diagnostics: vscode.Diagnostic[], diagnosticCode: DiagnosticCode
+): vscode.CodeAction => {
+  const detection = scanResultsService.getDetectionById(diagnosticCode.uniqueDetectionId);
+
+  const ignorePathAction = new vscode.CodeAction(
+      `open violation card for ${detection?.message}`,
```

Comment:
The same question here: do we want to be able to complete the action if detection is null?